File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    chk = [False]*(n+1)
    for x in reserve:
        if x in lost:
            lost.remove(x)
            continue
        chk[x] = True
    lost.sort()
    for x in lost:
        if x>1 and chk[x-1]:
            chk[x-1] = False
            continue
        if x<n and chk[x+1]:
            chk[x+1] = False
            continue
        n -= 1
    print(chk)
    answer = n
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        boolean isReserve[] = new boolean[32];
        boolean isLost[] = new boolean[32];
        for (int r: reserve) {
            isReserve[r] = true;
        }
        for (int l: lost) {
            isLost[l] = true;
            if (isReserve[l]) {
                isReserve[l] = false;
                isLost[l] = false;
            }
        }
        Arrays.sort(lost);
        for (int l: lost) {
            if (!isLost[l]) continue;
            if (isReserve[l-1]) {
                isReserve[l-1] = false;
                continue;
            } else if (isReserve[l+1]) {
                isReserve[l+1] = false;
                continue;
            }
            n -= 1;
        }
        answer = n;
        return answer;
    }
}